### PR TITLE
Disables auto-changelog and uses more defaults in goreleaser

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Changes from defaults should be commented or removed

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
I noticed our release pipeline is backwards. We release before e2e testing the binary we made. We should do "goreleaser build" then "make e2e" then "goreleaser release" I think

**Comment by mathetake** [comment]
> I noticed our release pipeline is backwards. We release before e2e testing the binary we made. We should do "goreleaser build" then "make e2e" then "goreleaser release" I think

I think this is intentional in order to test the "uploaded" binary works correctly vs testing the locally built binary..

**Comment by mathetake** [comment]
testing the locally built binary in the release workflow is basically a duplication of the e2e in the main branch's workflow..

File: .goreleaser.yaml [hunk 1]
```diff
@@ -25,26 +27,21 @@ builds:
...
+    tap:
       owner: tetratelabs
       name: homebrew-getenvoy
       # Use token other than the default one (which is limited in scope to
       # https://github.com/tetratelabs/getenvoy only)
       token: "{{ .Env.HOMEBREW_GETENVOY_GITHUB_TOKEN }}"
     folder: Formula
     homepage: https://github.com/tetratelabs/getenvoy
-    description: Distribute, deploy and operate a fleet of Envoys
+    description: Download and run Envoy
```
Comment (mathetake) [inline .goreleaser.yaml:42]: nit this is not in line with the "install" command 😄 
```suggestion
    description: Install and run Envoy
```

**Comment by codefromthecrypt** [comment]
I'll save the e2e on release refactoring for a different PR, but glad the conversation started. We should be testing the binary we built "before" releasing it ideally. ex "goreleaser build" only after e2e "goreleaser release". The only test we should need to do post-release is that the tar made as an attachment to the releases page. The tar would have exactly the same bytes as we tested.

**Comment by mathetake** [comment]
yeah maybe be testing shasum is enough for post release. Thanks.

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
thanks for the help again!

**Pull request merged**